### DISCUSSION TITLE
issue related to legacy parameters migration

### DISCUSSION THREAD

**Discussion by da00891219** [coder/coder#7956] [General] [answered]
I am referring to https://coder.com/docs/v2/latest/templates/parameters#migration

q1: "Migration is available until v0.24.0 (Jun 2023) release." - Legacy parameters are unsupported after this release?
q2: We have about 15-20 templates, what is your recommendation? rewrite the templates or use legacy parameters?
q3: The legacy parameters migration what we do now, wont work after v0.24.0 and you are saying to use "feature_use_managed_variables = "true"", so for all our future templates we must use this parameter?

**Comment by matifali** [comment, marked as answer]
> q1: "Migration is available until v0.24.0 (Jun 2023) release." - Legacy parameters are unsupported after this release. 

Yes, they are not supported any more in v0.24.0


> q2: We have about 15-20 templates, what is your recommendation? rewrite the templates or use legacy parameters? 

Use legacy parameters as given in the guide to migrate old workspaces without breaking once migrated you have to remove the legacy parameters and then you can safely upgrade Coder to v0.24.0. 

Another option is if you can delete all workspaces then you can just write a new template by using the coder_parameter and then create workspaces again using the updated template. 


> q3: The legacy parameters migration what we do now, wont work after v0.24.0 and you are saying to use "feature_use_managed_variables = "true"", so for all our future templates we must use this parameter?

You must set this if you want to use terraform `variable` for secrets or variables that are template scoped. For workspaces scoped parameters, you can only use `coder_parameter` data 


I hope it answers your questions. @da00891219

**Reply by da00891219** [reply]
1. v0.24.0 is released, how long you keep v0.24.0? When will be the next release in general? I doubt by next release legacy variables won't work and it is waste of time for us to upgrade templates now with legacy variable approach? Am I thinking correctly?

3. As v0.24.0 is already released and legacy variables supported until v0.24.0, what would be the best approach for us to upgrade templates? Writing new templates with coder_parameter data source? or writing templates with feature_use_managed_variables flag? Which is better for future?

**Reply by matifali** [reply]
Legacy parameters no longer work in release v0.24.0, the migration guide only applies if you are on an earlier release. 
`feature_use_managed_variables` is needed from now on if you want to use variable resources. 
You can only use them with this.
And yes you should use `coder_parameter` for all workspace-scoped variables.

**Reply by da00891219** [reply]
Can you please provide an example template with feature_use_managed_variables. And how to call them in the other resources like kubernetes_pod,

**Reply by matifali** [reply]
please check https://github.com/coder/coder/tree/main/examples/templates/fly-docker-image

**Reply by matifali** [reply]
Hi @da00891219 if you have successfully converted your templates. Please marke this discussion as resolved.
